crates/lib3h/src/engine/real_engine.rs

```diff
@@ -363,9 +365,11 @@ impl<T: Transport, D: Dht, SecBuf: Buffer, Crypto: CryptoSystem> RealEngine<T, D
         // First create DhtConfig for space gateway
         let agent_id = std::string::String::from_utf8_lossy(&join_msg.agent_id).into_owned();
         let this_net_peer = self.network_gateway.borrow().this_peer().clone();
+        let this_peer_transport =
+            Url::parse(format!("hc://{}", this_net_peer.peer_address.clone()).as_str()).unwrap();
```

Comment:
take out the slashes, making it a non-hierarchical url: `hc:{}`

Reply:
@neonphog Done, but I think this needs to be encapsulated better in a future PR.